code/modules/mob/dead/observer/observer.dm

```diff
@@ -815,18 +815,29 @@ This is the proc mobs get to turn into a ghost. Forked from ghostize due to comp
 		if (ghostimage)
 			client.images -= ghostimage //remove ourself
 
-mob/dead/observer/MayRespawn(var/feedback = 0, var/respawn_time = 0)
+mob/dead/observer/MayRespawn(var/feedback = 0, var/respawn_type = null)

```

Comment:
Bay condensed all the respawn checks into this central function. i have modified it